[SQL] RustError: table with column MAP<CUSTOMTYPE, INT> fails to compile

This is due to the custom types not implementing the `Ord` trait. 

SQL: 

```sql
create type ty0 as (
    id int,
    v varchar
);

create table t0 (id int, s MAP<ty0, int>);

create materialized view v1 as select id, s from t0;
```

Rust Error:

```rust
 error: could not compile `feldera_pipe_operator_a76b7c1fe6ec7ef3` (lib) due to 6 previous errors
```

```rust
 error[E0277]: the trait bound `struct_fca518ff8ddd2b74: Ord` is not satisfied
    --> crates/feldera_pipe_operator_a76b7c1fe6ec7ef3/src/lib.rs:119:32
     |
119  |                     .collect::<BTreeMap<struct_fca518ff8ddd2b74, Option<i32>>>())), 
     |                      -------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Ord` is not implemented for `struct_fca518ff8ddd2b74`
     |                      |
     |                      required by a bound introduced by this call
     |
     = help: the trait `FromIterator<(K, V)>` is implemented for `BTreeMap<K, V>`
     = note: required for `BTreeMap<struct_fca518ff8ddd2b74, std::option::Option<i32>>` to implement `FromIterator<(struct_fca518ff8ddd2b74, std::option::Option<i32>)>`
note: required by a bound in `collect`
    --> /Users/abhizer/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:1967:19
     |
1967 |     fn collect<B: FromIterator<Self::Item>>(self) -> B
     |                   ^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::collect`
help: consider annotating `struct_fca518ff8ddd2b74` with `#[derive(Ord)]`
     |
66   +     #[derive(Ord)]
67   |     pub struct struct_fca518ff8ddd2b74 {
     |
```

--- a/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/rust/ToRustInnerVisitor.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/rust/ToRustInnerVisitor.java
@@ -916,7 +916,7 @@ public class ToRustInnerVisitor extends InnerVisitor {
     }
 
     void generateStructDeclaration(DBSPTypeStruct struct) {
-        this.builder.append("#[derive(Clone, Debug, Eq, PartialEq, Default)]")
+        this.builder.append("#[derive(Clone, Debug, Eq, PartialEq, Default, PartialOrd, Ord)]")
                 .newline();
         builder.append("pub struct ")
                 .append(Objects.requireNonNull(struct.sanitizedName))

--- a/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/CatalogTests.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/CatalogTests.java
@@ -613,4 +613,16 @@ public class CatalogTests extends BaseSQLTests {
                 create materialized view V as SELECT * FROM T;""";
         this.getCCS(sql);
     }
+
+    @Test
+    public void issue3892() {
+        String sql = """
+                create type ty0 as (
+                    id int,
+                    v varchar
+                );
+                create table t0 (id int, s MAP<ty0, int>);
+                create materialized view v1 as select id, s from t0;""";
+        this.getCCS(sql);
+    }
 }

